ICSharpCode.Decompiler/CSharp/ExpressionBuilder.cs

```diff
@@ -2272,7 +2270,9 @@ TranslatedExpression TranslateArrayInitializer(Block block)
 					container.Peek().Elements.Add(aie);
 					container.Push(aie);
 				}
+				astBuilder.UseSpecialConstants = !type.IsCSharpPrimitiveIntegerType();
 				var val = Translate(value, typeHint: type).ConvertTo(type, this, allowImplicitConversion: true);
+				astBuilder.UseSpecialConstants = true;
```

Comment:
Are you sure the previous value was `true`? e.g. in nested array initializers...